staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -248,13 +248,25 @@ func (j *JSONPath) evalArray(input []reflect.Value, node *ArrayNode) ([]reflect.
 		}
 		if params[0].Value < 0 {
 			params[0].Value += value.Len()
+
+			// if index is still negative, return an
+			// out of bounds error...
+			if params[0].Value < 0 {
```

Comment:
We are doing some validation down bellow (line 261 of old file). For both of these checks, should we just combine the new checks with those old checks?

```go
 			if params[0].Value >= sliceLength || params[0].Value < 0 {
 				return input, fmt.Errorf("array index out of bounds: index %d, length %d", params[0].Value, sliceLength)
 			}
 			if params[1].Value > sliceLength || params[1].Value < 0 {
 				return input, fmt.Errorf("array index out of bounds: index %d, length %d", params[1].Value-1, sliceLength)
 			}
```